### DISCUSSION TITLE
system stability

### DISCUSSION THREAD

**Discussion by CalvinKoung** [bitcraze/discussions#1211] [Q&A]
I would like to ask if anyone in the community has encountered the same problem. My crazyflie is quite unstable when flying under the UWB system. Sometimes it can complete the path smoothly, and sometimes it crashes halfway through. Is it because of battery consumption? Or is it the original factor of the system? I've turned off all other sources that might be blocking the signal, but still have the same problem. Thanks

**Comment by ataffanel** [comment]
Hi, Th Loco system is not supposed to be as accurate than a Mocap or the Lighthouse system for example, so some noise is expected. However, there should not be big inconsistency between flight and definitely no inconsistent crashes.

Can you tell us a bit more how your system is setup, what are you using to control your Crazyflies, what kind of trajectories are you using and if you have any other decks on the Crazyflie, and what version of firmware/software you are using?

Among the cause of crash we have seen, there is indeed the battery but it is for quite aggressive flight that pushed the Crazyflie to the limit. One quite common cause of random crash has also been mechanical one like broken motor mount or deck that disconnects from the pins.

**Reply by CalvinKoung** [reply]
Hi, I use high_level_command with python to control my crazyflie equipped with UWB deck, Passive marker deck, battery, and flow deck V2, and my flight path is simple(After taking off, circle a rectangle at a height of one meter, then circle a rectangle at a height of two meters, and finally land gradually vertically.).   thanks

**Reply by ataffanel** [reply]
Thanks for the extra information. What behavior does it have when it does not fly well? Does it look noisy or does it fly in the wrong direction? A short video of the Crazyflie flying well and not well could help understanding what could be happening.

One things that comes to mind, is that in the Loco system, it is very important for the Crazyflie to be switched ON in the direction facing X. If this is not done, the Crazyflie will fly in the wrong direction until ti can recover its YAW orientation (witch takes too long, so it will crash before ...)

**Reply by CalvinKoung** [reply]
I have some video and picture of anchor position.  The first video can completely fly 90%, but the second video crash at the beginning. 

![IMG_5660](https://github.com/bitcraze/discussions/assets/118436188/a7fcea6d-c963-439c-8c63-b504227cdc06)


https://github.com/bitcraze/discussions/assets/118436188/6c98d55f-9c9e-44e0-9212-53d5e16e1412


https://github.com/bitcraze/discussions/assets/118436188/d4fef485-59d9-4855-8af1-a31629303390

**Reply by ataffanel** [reply]
Hi, Thanks for the videos!

This does not look like a positioning problem but something else. A couple of things we can thing about:
 - The motor connection of one of the motor might be bad and glitchy
 - Something might block one of the propeller, like the battery wire.
 - One propeller flying away also generates very similar crash.
 - If this is not a stock firmware, you might have a bug in your firmware.
 - If it is a positioning problem, it would be a very big outlier.

Essentially, this looks way too violent to be caused by a positioning problem. One way to rule out positioning would be to log the position estimation while flying and see if there is any big outlers.  There are filters in the Crazyflie that should prevent that to happen though. So my advice would be to check your hardware to see that everything is in good order, maybe to switch to newer propeller.

**Reply by CalvinKoung** [reply]
Hi, Why do you infer that the problem is with the propeller ? I flash the latest crazyflie firmware (2024.2) but incountering the same problems. I can provide you with the flight path recorded by vicon.

These were two successful flights, but it can be seen that the stability is still not good.
<img width="803" alt="截圖 2024-03-09 00 30 31" src="https://github.com/bitcraze/discussions/assets/118436188/8e0a8863-2f31-4cad-9610-edab16c507de">

**Reply by knmcguire** [reply]
So the reason that we say it is either a propeller or motor problem, is just out of experience of looking at many, many crazyflie crashes over the years. If it is an instability problem, it would amplitude of the wobble would increase until its no longer controllable. If it is a positioning (lose) problem, the crazyflie would drift away to the side. Flying around and suddenly flipping is usually due to a hardware issue caused by the propeller, motor or sudden interruption of the motor or propeller due to some lose wire or such.


You could for instance check these things out and rule out most issues with the motor/battery/propeller. On the console tab of the cfclient, we have a couple of tests that you can try out, like the motor/propeller test (place it flat on a table first) or the battery test. Then at least you would be able to rule those 2 out as the cause of the flipping

![image](https://github.com/bitcraze/discussions/assets/11061260/39876046-3deb-48d3-96eb-cbc52b649dea)


Mind that the loco positioning system is not a milimeter precision positioning system, and the regular flights we see on your videos are not what we consider out of the ordinary considering the lab you are flying in. If you'd like to improve the flight perhaps it is good to check out the robuust mode here described in this blogpost: https://www.bitcraze.io/2021/06/robust-kalman-filter-for-ultra-wideband-localization/

**Reply by CalvinKoung** [reply]
Hi, it my propeller test result. 
![VirtualBox_BitcrazeVM 1_12_03_2024_14_46_10](https://github.com/bitcraze/discussions/assets/118436188/8c26cef9-e624-43b9-a21a-c78c4e959e87)

**Reply by knmcguire** [reply]
Those variances looks really bad actually. That is maybe due crashes, that the props are definitely not in good condition anymore. It could also be due to a bad motor or bend axis. 

Do you have spare motors and props to change some of these? Or exchange the motors and props from another crazyflie that is flying well.
